src/reducers/index.js

```diff
@@ -51,60 +38,54 @@ function reducer(state = {}, action) {
     }
 
     case TOGGLE_NODE_HOVERED: {
-      return Object.assign({}, state, {
+      return updateState({
         nodeHovered: action.nodeHovered
       });
     }
 
-    case TOGGLE_PARAMETERS: {
-      const paramIDs = state.nodes.filter(id => state.nodeIsParam[id]);
-      return Object.assign({}, state, {
-        nodeDisabled: paramIDs.reduce(
-          (disabled, id) =>
-            Object.assign({}, disabled, {
-              [id]: !action.parameters
-            }),
-          state.nodeDisabled
-        ),
-        parameters: action.parameters
-      });
-    }
-
     case TOGGLE_TEXT_LABELS:
-      return Object.assign({}, state, {
+      return updateState({
         textLabels: action.textLabels
       });
 
     case TOGGLE_TAG_ACTIVE: {
-      return Object.assign({}, state, {
+      return updateState({
         tagActive: Object.assign({}, state.tagActive, {
           [action.tagID]: action.active
         })
       });
     }
 
     case TOGGLE_TAG_FILTER: {
-      return Object.assign({}, state, {
+      return updateState({
         tagEnabled: Object.assign({}, state.tagEnabled, {
           [action.tagID]: action.enabled
         })
       });
     }
 
     case TOGGLE_THEME: {
-      return Object.assign({}, state, {
+      return updateState({
         theme: action.theme
       });
     }
 
+    case TOGGLE_TYPE_DISABLED: {
+      return updateState({
+        typeDisabled: Object.assign({}, state.typeDisabled, {
+          [action.typeID]: action.disabled
+        })
+      });
```

Comment:
The object assign here makes it difficult for me to understand what you want to update, could it change to something like (or is it because you've used object assign in updateState)

```
case TOGGLE_TYPE_DISABLED: {
      return updateState({
        typeDisabled: {
          ...state.typeDisabled,
          [action.typeID]: action.disabled
        }
      })
```

Reply:
I've tried to keep to using Object.assign over spread in the reducer. Honestly, I find Object.assign slightly easier to read, but I think it doesn't matter much either way? Seeing as that's what I've used throughout the rest of the reducer, and it's more common to use it in reducers generally, I'd prefer to keep it as is for consistency.

Reply:
Cool cool - maybe for me it slightly breaks the focus when reading a case when the updateState has been moved to the top (mental model for nesting object assigns located in different locations)